File: backend/src/main/java/com/bidbridge/backend/entity/LoadingStyle.java

package com.bidbridge.backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;
import java.time.Instant;

@Data
@NoArgsConstructor
@Entity
@Table(name = "loading_styles", schema = "public")
public class LoadingStyle {

    @Id
    @GeneratedValue
    private UUID id;

    private String name;

    @ManyToOne
    @JoinColumn(name = "company_id")
    private Company company;

    private Instant createdAt;
}

File: frontend/src/types/loading-style.ts
export interface LoadingStyle {
  id: string;
  name: string;
  company_id: string;
  company_name: string;
  created_at: string;
}

export interface LoadingStyleApiResponse {
  id: string;
  name: string;
  companyId: string;
  companyName: string;
  createdAt: string;
}

export function transformLoadingStyleFromApi(
  apiLoadingStyle: LoadingStyleApiResponse
): LoadingStyle {
  return {
    id: apiLoadingStyle.id,
    name: apiLoadingStyle.name,
    company_id: apiLoadingStyle.companyId,
    company_name: apiLoadingStyle.companyName,
    created_at: apiLoadingStyle.createdAt,
  };
}
